joomla/joomla-cms
# [4.0] Fix fatal error for fake URL in com_menus frontend

Pull Request for Issue #32252.

### Summary of Changes
This PR fixes issue #32252. From what I see, we only uses com_menus in the frontend (am I correct?) to display menu items in modal popup only, so we should limit access to all other views (accept view items, layout modal), all other requests are not allowed.

I also remove the Display controller (we uses backend code, so controller is not needed here

### Testing Instructions
# Test 1

1. See https://github.com/joomla/joomla-cms/issues/32252 , confirm the issue
2. Apply patch, try to access to that fake URL again. Confirm that the fatal error is fixed (the system now displaying The requested page can't be found)

#Test 2: Make sure you can still select a menu item in the frontend

- Create a menu item to link to **Create Article** menu item type
- Access to that menu item to submit article (You need to login using a super admin account or an account has permission to create article first)
- Look at the editor, Select CMS Content -> Menu, make sure a popup is displayed to allow you to insert a menu item in the article.
